source/agora/consensus/EnrollmentManager.d

```diff
@@ -1091,3 +1105,115 @@ unittest
         }
     }
 }
+
+/// tests for `EnrollmentManager.getValidatorCount
+unittest
+{
+    import agora.common.Amount;
+    import agora.consensus.data.Transaction;
+    import agora.consensus.Genesis;
+    import std.algorithm;
+    import std.format;
+    import std.conv;
+
+    scope storage = new TestUTXOSet;
+
+    auto gen_key_pair = getGenesisKeyPair();
+    KeyPair key_pair = KeyPair.random();
+
+    foreach (idx; 0 .. 8)
+    {
+        auto input = Input(hashFull(GenesisTransaction), idx.to!uint);
+
+        Transaction tx =
+        {
+            TxType.Freeze,
+            [input],
+            [Output(Amount.MinFreezeAmount, key_pair.address)]
+        };
+
+        auto signature = gen_key_pair.secret.sign(hashFull(tx)[]);
+        tx.inputs[0].signature = signature;
+        storage.put(tx);
+    }
+
+    // create an EnrollmentManager object
+    auto man = new EnrollmentManager(":memory:", key_pair);
+    scope (exit) man.shutdown();
+    Hash[] utxo_hashes = storage.keys;
+
+    Enrollment[] enrollments;
+    Enrollment enroll1;
+    Enrollment enroll2;
+    Enrollment enroll3;
+    ulong block_height = 2;
+
+    // create and add the first Enrollment object
+    auto utxo_hash1 = utxo_hashes[0];
+    assert(man.createEnrollment(utxo_hash1, block_height, enroll1));
+    assert(!man.hasEnrollment(utxo_hash1));
+    assert(man.add(enroll1, &storage.findUTXO));
+    assert(man.enroll_pool.count() == 1);
+    assert(man.hasEnrollment(utxo_hash1));
+    enrollments ~= enroll1;
+    assert(man.getValidatorCount(block_height) + enrollments.length == 1);
+
+    man.clearExpiredValidators(block_height);
+    foreach (ref enrollment; enrollments)
+        man.addValidator(enrollment, block_height, &storage.findUTXO);
+
+    enrollments.length = 0;
+    block_height = 3;
+
+    // create and add the second Enrollment object
+    auto utxo_hash2 = utxo_hashes[1];
+    assert(man.createEnrollment(utxo_hash2, block_height, enroll2));
+    assert(!man.hasEnrollment(utxo_hash2));
+    assert(man.add(enroll2, &storage.findUTXO));
+    assert(man.enroll_pool.count() == 1);
+    assert(man.hasEnrollment(utxo_hash2));
+    enrollments ~= enroll2;
+    assert(man.getValidatorCount(block_height) + enrollments.length == 2);
+
+    man.clearExpiredValidators(block_height);
+    foreach (ref enrollment; enrollments)
```

Comment:
The `enrollments` in all the `foreach` statement always has just one element, so why don't you use just the variable of `Enrollment` type and not to use `foreach` statement?

Reply:
I thought after watching ClearExpiredValidators.
Please explain in detail what the problem is.
ClearExpiredValidators :
enrolled_height <= block_height - Enrollment.ValidatorCycle
getValidatorCount:
enrolled_height >= block_height - Enrollment.ValidatorCycle + 1

Reply:
I changed it.
Thank you for your good suggestion.